z_infra_provisioning/cloud_infra_center/ocp_upi/roles/configure-pre-check/tasks/main.yaml

```diff
@@ -87,4 +87,23 @@
 - name: Check server group quota
   fail:
     msg: "Server group number is not enough, current: {{ server_group.stdout_lines[0] }}, required: 1"
-  failed_when: "server_group.stdout_lines[0]|int < 1"
\ No newline at end of file
+  failed_when: "server_group.stdout_lines[0]|int < 1"
+
+- name: Get storage provide
+  shell: openstack volume service list -f value | grep cinder-volume |awk '{print$4$5}'
+  register: storage_backend
+  when: use_storage_provider == true
+
+- name: Check if storage provide is configured correctly
+  fail:
+    msg: "Storage provider is not enabled and active."
+  failed_when: 
+  - storage_backend.stdout_lines[0] != 'enabledup'
+  - use_storage_provider == true
+
+- name: Check volume template is properly set
```

Comment:
```
[root@m5404019 ~]# openstack volume type list
+--------------------------------------+-----------------------------------------+-----------+
| ID                                   | Name                                    | Is Public |
+--------------------------------------+-----------------------------------------+-----------+
| 9cd18c99-ddba-4d4e-b4f5-658567dc2d47 | new_storage_provider_v7k base template  | True      |
| e743ba62-d32d-43a2-ade1-14fd1af9a4bf | new_storage_provider_ds8k base template | True      |
+--------------------------------------+-----------------------------------------+-----------+
[root@m5404019 ~]#
[root@m5404019 ~]# openstack volume service list -f value | grep cinder-volume |awk '{print$4$5}'
Groupenabled
Groupenabled
```

whether we are using different env ? should this be $5$6 instead at line 93?

Reply:
No, Umm, I suppose this this command to valiidate the storage provider is configured and active, so I just check the volume service for column 4 and 5,  my test stand returns `enableup`, but perhaps that may change in different build and env. I will double check it later.